app/assets/javascripts/float-kit/addon/components/d-menu.gjs

```diff
@@ -25,6 +25,11 @@ export default class DMenu extends Component {
   registerTrigger = modifier((element) => {
     this.menuInstance.trigger = element;
     this.options.onRegisterApi?.(this.menuInstance);
+
+    return () => {
+      this.menuInstance.destroy();
+      this.menuInstance = null;
+    };
```

Comment:
I noticed we were not correctly calling destroy when the menu was removed from DOM.
Im not sure we need to nullify it, but it seems like an appropriate time to do it.